Unity-Technologies/ScriptableRenderPipeline
# GC fix (#1227490)

### Purpose of this PR
Fixed GC pressure in `VolumeParameter<T>.GetHashCode()` in CoreRP. `ReferenceEquals()` will box value types so it's better to use the default `EqualityComparer` as it's properly typed.

This fixes fogbugz#1227490 and affects both URP and HDRP (in the sense that it will reduce GC pressure for both of these pipelines).

Not sure which QA to assign to this PR so I'll let the reviewers choose.

---
### Testing status

Issue only reproduces in standalone player. Editor is fine.

How to reproduce:
    - Open URP GraphicsTestScene.
    - Build Project containing scene 104 BokehDepthOfField
    - Build with Deep Profiling enabled.
    - Check profiler, see that VolumeParameter.GetHashCode is not allocating memory anymore.
